### PR TITLE
enhancement(observability): API batch support + tests

### PR DESCRIPTION
**Pull request by leebenson**
Follow-up to @sghall's #5002 to add support for `*Batch` subscriptions in the API client, and some tests to assert batching is happening as expected. I also realised I neglected to add a test for the previous `bytesProcessedTotal` and component variants, which have been added here.

No tests for error metrics yet, since those aren't typically thrown. Will probably tackle after #4925 is solved.

### CODE REVIEW

**Comment by sghall** [comment]
Nice @leebenson.  I'll look at this today.  Want to spend some time to really understand the test setup and patterns here. 👍

File: lib/vector-api-client/src/gql/metrics.rs [hunk 1]
```diff
@@ -110,4 +154,16 @@ impl MetricsSubscriptionExt for crate::SubscriptionClient {
 
         self.start::<ComponentBytesProcessedTotalSubscription>(&request_body)
     }
+
+    /// Executes a components events processed total metrics subscription
```
Comment (sghall) [inline lib/vector-api-client/src/gql/metrics.rs:158]: nit:  

/// Executes a components **bytes** processed total batch metrics subscription

File: tests/api.rs [hunk 2]
```diff
@@ -388,8 +388,169 @@ mod tests {
...
+                    .unwrap()
+                    .data
+                    .unwrap()
+                    .component_bytes_processed_total,
+            );
+        }
+
+        // Should only show byte totals on sinks
+        assert_eq!(map[&0].name, "bytes_processed_total_sink");
+        assert_eq!(map[&0].name, "bytes_processed_total_sink");
```
Comment (sghall) [inline tests/api.rs:501]: nit:  duplicate assertion here

**Review by sghall — APPROVED** [review]
LGTM 👍

File: tests/api.rs (hunk 2 above)
Comment (leebenson) [reply tests/api.rs:501]: Thanks! That second one should pulling from `map[&1].name`.

**Pull request merged**